ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activityEmailInput.component.ts

```diff
@@ -66,63 +73,82 @@ export class ActivityEmailInput implements OnChanges {
         }
     }
 
-    public onChange(answer: string): void {
-        const allFieldsTouched = Object.values(this.emailForm.controls).every(control => control.touched);
-        if (this.emailForm.valid) {
-            this.saveAnswer(answer);
-        } else if (allFieldsTouched) {
-            this.saveAnswer(null);
-        }
-    }
-
     public fieldHasError(field: string, error: string): boolean {
-        return this.emailForm.touched && this.emailForm.controls[field].errors && this.emailForm.controls[field].errors[error];
+        return this.emailForm && this.emailForm.touched && this.emailForm.controls[field].errors
+            && this.emailForm.controls[field].errors[error];
     }
 
     private initEmailForm(): void {
+        interface EmailForm {
+            email: string;
+            confirmEmail?: string;
+        }
+        // can't use the Angular email validator as does not match Pepper back-end email validator
+        const EmailPatternValidator = Validators.pattern(EMAIL_REGEXP);
+
         if (this.block.confirmEntry) {
+
             this.emailForm = this.formBuilder.group({
                 email: new FormControl({
                     value: this.block.answer,
                     disabled: this.readonly
-                }, Validators.email),
+                }, EmailPatternValidator),
                 confirmEmail: new FormControl({
-                    value: this.block.answer,
+                    value: this.block.confirmationValue,
```

Comment:
We can save on the backend only valid answer(email), so, we don't need to store confirmationValue in the block and we can show the answer in both fields.